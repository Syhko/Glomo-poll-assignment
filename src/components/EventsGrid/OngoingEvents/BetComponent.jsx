import React from "react";
import "./style.scss";

const OngoingEvents = ({ playerOne, playerTwo }) => {
  return (
    <div className="bet-component-wrapper">
      <p className="bet-player">{playerOne}</p>
      <p className="versus">VS</p>
      <p className="bet-player">{playerTwo}</p>
    </div>
  );
};

export default OngoingEvents;
