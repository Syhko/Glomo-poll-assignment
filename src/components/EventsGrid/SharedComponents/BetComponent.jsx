import React from "react";
import "./style.scss";

const OngoingEvents = ({ playerOne, playerTwo, id, group }) => {
  const [isBetting, setIsBetting] = React.useState(false);
  // getting the data from local storage to populate the component
  const [betSelected, setBetSelected] = React.useState(
    localStorage.getItem(id) === undefined ? "" : localStorage.getItem(id)
  );
  // the bet saving system with local storage
  const betFunction = decision => {
    setBetSelected(decision);
    localStorage.setItem(id, decision);
  };

  return (
    <div
      className="bet-component-wrapper"
      onMouseEnter={() => setIsBetting(true)}
      onMouseLeave={() => setIsBetting(false)}
    >
      <div className="group-label">
        <p>{group}</p>
      </div>
      {!isBetting && (
        <>
          <p
            className={
              betSelected === "Home" ? "bet-player-selected" : "bet-player"
            }
          >
            {playerOne}
          </p>
          {betSelected === "Draw" && (
            <p className="bet-player-selected">Draw</p>
          )}
          <p
            className={
              betSelected === "Away" ? "bet-player-selected" : "bet-player"
            }
          >
            {playerTwo}
          </p>
        </>
      )}
      {isBetting && (
        <>
          <p className="bet" onClick={() => betFunction("Home")}>
            Home
          </p>
          <p className="bet" onClick={() => betFunction("Draw")}>
            Draw
          </p>
          <p className="bet" onClick={() => betFunction("Away")}>
            Away
          </p>
        </>
      )}
    </div>
  );
};

export default OngoingEvents;
