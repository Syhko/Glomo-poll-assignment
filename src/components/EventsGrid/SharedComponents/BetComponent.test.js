import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import BetComponent from "./BetComponent";
import "../../../setupTests";

it("BetComponent should render with given props", () => {
  const playerOne = "player-one";
  const playerTwo = "player-two";
  const id = 10;
  const group = "group";
  const component = shallow(
    <BetComponent
      playerOne={playerOne}
      playerTwo={playerTwo}
      id={id}
      group={group}
      debug
    />
  );
  expect(component).toMatchSnapshot();
});
