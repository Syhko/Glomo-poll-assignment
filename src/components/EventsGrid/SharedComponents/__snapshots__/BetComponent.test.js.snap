// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetComponent should render with given props 1`] = `ShallowWrapper {}`;
